chaco/tools/pan_tool.py

```diff
@@ -170,6 +170,11 @@ def panning_mouse_move(self, event):
         for direction, bound_name, index in direction_info:
             if not self.constrain or self.constrain_direction == direction:
                 mapper = getattr(plot, direction + "_mapper")
+
+                # there is nowhere one could pan!
+                if mapper._null_screen_range or mapper._null_data_range:
```

Comment:
This is where we start getting weird things (at least for the PanTool), so I try to catch it early.  _Unfortunately_ this trait is not defined on the `AbstactMapper` base class, so I am uncertain if this is safe here.  However, it is defined on all of the `LinearMapper`, `LogMapper` and `PolarMapper` classes.

Ultimately we are ending up in a situation where `newlow = [-inf]`, `newhigh = [-inf]`, `domain_min = -inf`, domain_high = inf`.  I do not understand why `LinearMapper.map_data` returns a singleton array of the low end of the range when one of the null traits is true, see:
https://github.com/enthought/chaco/blob/fdd858aa6dbc76addb50d011fb81e879ce8e0355/chaco/linear_mapper.py#L60-L63

This reminds me of #528 / #589 
I feel like we should not change the types of the return value for a function for special cases, or if we do make it cleary documented that that is an edge case and handle it appropriately.  What happens here is that `array([-inf])` gets carried through and tried to be handled as if it were a normal float representing a a point in data space.  But there is no range of data space at all in this scenario

Also there is this comment on `AbstractMapper`: https://github.com/enthought/chaco/blob/fdd858aa6dbc76addb50d011fb81e879ce8e0355/chaco/abstract_mapper.py#L19 
but `domain_limits` very much appear to be being used here. (Not super important just pointing it out)


Overall I am unsure we want to use these traits here, but we definitely need some sort of edge `nan`, `inf`, `None`, etc case checking here and in other tools.

Reply:
Note, the change from the original commit of this PR seems to solve the reported issue, but I wanted to investigate up the chain to see where things were coming from and catch it earlier